soda/core/soda/scan.py

```diff
@@ -481,7 +526,7 @@ def execute(self) -> int:
             exit_value = 3
             self._logs.error(f"Error occurred while executing scan.", exception=e)
         finally:
-
+            self.scan_results = self.build_scan_results(scan=self)
```

Comment:
set_data_source_name is a method to configure the scan called before the execute.  i don't think the build scan results belongs here.

Let us know what you want to accomplish and we can help you find the most appropriate and fastest way to do it.